stan/math/prim/mat/fun/gp_dot_prod_cov.hpp

```diff
@@ -1,14 +1,18 @@
 #ifndef STAN_MATH_PRIM_MAT_FUN_COV_DOT_PROD_HPP
 #define STAN_MATH_PRIM_MAT_FUN_COV_DOT_PROD_HPP
 
+#include <boost/utility/enable_if.hpp>
```

Comment:
Not used, so get rid of it. I don't think is_vector_like, check_nonnegative, or is_constant are used either.